src/main/java/org/opensearch/flowframework/workflow/DeleteAgentStep.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.delete.DeleteResponse;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.util.ParseUtils;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static org.opensearch.flowframework.common.CommonValue.AGENT_ID;
+
+/**
+ * Step to delete a agent for a remote model
+ */
+public class DeleteAgentStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(DeleteAgentStep.class);
+
+    private MachineLearningNodeClient mlClient;
+
+    static final String NAME = "delete_agent";
+
+    /**
+     * Instantiate this class
+     * @param mlClient Machine Learning client to perform the deletion
+     */
+    public DeleteAgentStep(MachineLearningNodeClient mlClient) {
+        this.mlClient = mlClient;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(
+        String currentNodeId,
+        WorkflowData currentNodeInputs,
+        Map<String, WorkflowData> outputs,
+        Map<String, String> previousNodeInputs
+    ) throws IOException {
+        CompletableFuture<WorkflowData> deleteAgentFuture = new CompletableFuture<>();
+
+        ActionListener<DeleteResponse> actionListener = new ActionListener<>() {
+
+            @Override
+            public void onResponse(DeleteResponse deleteResponse) {
+                deleteAgentFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(Map.entry("agent_id", deleteResponse.getId())),
```

Comment:
should we use `WorkflowResources.getResourceByWorkflowStep("register_agent")` or still add it to the enum just for the fetching constant process. Just in terms of less places we type even something like `"agent_id"`

Reply:
Sure, but we should do the same for delete connector, delete model, and unload model.

This class is a literal copy/paste of the other two delete classes, whatever we change here should match there.  Can we go back in another PR to refactor all of these?